sormas-ui/src/main/java/de/symeda/sormas/ui/events/eventParticipantsLineListing/layout/LineListingLayout.java

```diff
@@ -0,0 +1,360 @@
+package de.symeda.sormas.ui.events.eventParticipantsLineListing.layout;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import com.vaadin.data.Binder;
+import com.vaadin.data.BinderValidationStatus;
+import com.vaadin.icons.VaadinIcons;
+import com.vaadin.ui.Alignment;
+import com.vaadin.ui.Button;
+import com.vaadin.ui.ComboBox;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.TextField;
+import com.vaadin.ui.VerticalLayout;
+import com.vaadin.ui.Window;
+import com.vaadin.ui.themes.ValoTheme;
+
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.caze.CaseDataDto;
+import de.symeda.sormas.api.event.EventDto;
+import de.symeda.sormas.api.event.EventParticipantDto;
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.infrastructure.district.DistrictReferenceDto;
+import de.symeda.sormas.api.infrastructure.region.RegionReferenceDto;
+import de.symeda.sormas.api.person.PersonDto;
+import de.symeda.sormas.api.utils.ValidationRuntimeException;
+import de.symeda.sormas.ui.UserProvider;
+import de.symeda.sormas.ui.utils.ButtonHelper;
+import de.symeda.sormas.ui.utils.CssStyles;
+import de.symeda.sormas.ui.utils.FieldHelper;
+import de.symeda.sormas.ui.utils.components.linelisting.line.DeleteLineEvent;
+import de.symeda.sormas.ui.utils.components.linelisting.line.LineLayout;
+import de.symeda.sormas.ui.utils.components.linelisting.model.LineDto;
+import de.symeda.sormas.ui.utils.components.linelisting.person.PersonField;
+import de.symeda.sormas.ui.utils.components.linelisting.person.PersonFieldDto;
+import de.symeda.sormas.ui.utils.components.linelisting.section.LineListingSection;
+
+public class LineListingLayout extends VerticalLayout {
+
+	private final ComboBox<RegionReferenceDto> region;
+	private final ComboBox<DistrictReferenceDto> district;
+
+	private final EventDto eventDto;
+	private final List<EventParticipantLineLayout> eventParticipantLines;
+	private final Window window;
+	private Consumer<LinkedList<LineDto<EventParticipantDto>>> saveCallback;
+
+	public LineListingLayout(Window window, EventDto eventDto) {
+		this.window = window;
+
+		setSpacing(false);
+
+		LineListingSection sharedInformationComponent = new LineListingSection(Captions.lineListingSharedInformation);
+
+		this.eventDto = eventDto;
```

Comment:
please move this up below `this.window` initialization

Reply:
changed